src/state/machines/turn.ts

```diff
@@ -725,6 +725,9 @@ export const TurnMachine = setup({
     },
     checkingEventConditions: {
       initial: "main",
+      on: {
```

Comment:
I tried moving this event to the top level "turn" state, but this allows this button to be pressed always, meaning you can keep on pressing it until you have a card draw you are happy with before actually ending the turn. The simplest solution I came up with now is that ending turn is essentially allowed in two states:

- Buying
- Using ability

The reason why it has also been added to event conditions is to prevent the button from flashing when putting cards in play (because this transitions to checking event conditions before #turn.buying)

Reply:
OK. So we now explicitly stating in which states endTurn is possible. This approach definitely has it's benefits.

Reply:
Yes exactly, yeah I think this is better than writing a bunch of guards and logic of when the turn should actually be able to be ended